impress-org/givewp
# Donors now only ever see their own donation history

Resolves #5675 

## Description

This PR improves error handling within the Donations repository, so that a donor is only ever shown their own donation history.

## Affects

This PR affects some frontend rendering logic for the donation history tab, but primarily impacts logic in the Donations repository.

## Visuals

<img width="949" alt="Screen Shot 2021-03-08 at 10 35 26 AM" src="https://user-images.githubusercontent.com/5186078/110343080-1ec96a00-7ffa-11eb-8471-e004f79dca12.png">

## Testing Instructions

To easily test the frontend behavior without needing to corrupt a database, you can comment out if statement at line 53 in DonorProfiles/Tabs/DonationHistory/DonationsRoute.php. This will force an error response.

1) View the donor profile app
2) Check that an error appears in the dashboard
3) Check that an error appears in the Donation History tab

Uncomment the line (returning it to it's normal state).

1) View the donor profile app
2) Check that the dashboard works correctly
3) Check that the Donation History tab works correctly

## Pre-review Checklist
